tests/resp_test.go

```diff
@@ -0,0 +1,176 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/dicedb/dice/core"
+)
+
+func TestSimpleStringDecode(t *testing.T) {
+	cases := map[string]string{
+		"+OK\r\n": "OK",
+	}
+	for k, v := range cases {
+		p := core.NewRESPParser(bytes.NewBuffer([]byte(k)))
+		value, err := p.DecodeOne()
+		if err != nil {
+			t.Error(err)
+			t.Errorf("error while decoding: %v", k)
+		}
+		if v != value {
+			t.Fail()
+		}
+	}
+}
+
+func TestError(t *testing.T) {
+	cases := map[string]string{
+		"-Error message\r\n": "Error message",
+	}
+	for k, v := range cases {
+		p := core.NewRESPParser(bytes.NewBuffer([]byte(k)))
+		value, err := p.DecodeOne()
+		if err != nil {
+			t.Error(err)
+			t.Errorf("error while decoding: %v", k)
+		}
+		if v != value {
+			t.Fail()
+		}
+	}
+}
+
+func TestInt64(t *testing.T) {
+	cases := map[string]int64{
+		":0\r\n":    0,
+		":1000\r\n": 1000,
+	}
+	for k, v := range cases {
+		p := core.NewRESPParser(bytes.NewBuffer([]byte(k)))
+		value, err := p.DecodeOne()
+		if err != nil {
+			t.Error(err)
+			t.Errorf("error while decoding: %v", k)
+		}
+		if v != value {
+			t.Fail()
+		}
+	}
+}
+
+func TestBulkStringDecode(t *testing.T) {
+	cases := map[string]string{
+		"$5\r\nhello\r\n": "hello",
+		"$0\r\n\r\n":      "",
+	}
+	for k, v := range cases {
+		p := core.NewRESPParser(bytes.NewBuffer([]byte(k)))
+		value, err := p.DecodeOne()
+		if err != nil {
+			t.Error(err)
+			t.Errorf("error while decoding: %v", k)
+		}
+		if v != value {
+			t.Fail()
+		}
+	}
+}
+
+func TestArrayDecode(t *testing.T) {
+	cases := map[string][]interface{}{
+		"*0\r\n":                                                   {},
+		"*2\r\n$5\r\nhello\r\n$5\r\nworld\r\n":                     {"hello", "world"},
+		"*3\r\n:1\r\n:2\r\n:3\r\n":                                 {int64(1), int64(2), int64(3)},
+		"*5\r\n:1\r\n:2\r\n:3\r\n:4\r\n$5\r\nhello\r\n":            {int64(1), int64(2), int64(3), int64(4), "hello"},
+		"*2\r\n*3\r\n:1\r\n:2\r\n:3\r\n*2\r\n+Hello\r\n-World\r\n": {[]int64{int64(1), int64(2), int64(3)}, []interface{}{"Hello", "World"}},
+	}
+	for k, v := range cases {
+		p := core.NewRESPParser(bytes.NewBuffer([]byte(k)))
+		value, err := p.DecodeOne()
+		if err != nil {
+			t.Error(err)
+			t.Errorf("error while decoding: %v", v)
+		}
+		array := value.([]interface{})
+		if len(array) != len(v) {
+			t.Fail()
+		}
+		for i := range array {
+			if fmt.Sprintf("%v", v[i]) != fmt.Sprintf("%v", array[i]) {
```

Comment:
@arpitbbhayani instead of using `fmt.Sprintf` to check for equality, should we instead use `reflect.DeepEqual` to have more precise checks?
With `fmt.Sprintf`, `"1"` and `1` would be considered equal.

Reply:
@asutosh97 and @rohanverma94 I tried changing the equality check, but the overall logic becomes super complex given that the unit test is testing against an array that can hold heterogeneous types of data. Hence I was relying on the just blunt serialization of it/

Given that this is just a correctness test I was planning to park it and create an issue for tracking it at a later stage.

It would be great if both of you can have a look at my PR. I have updated the PR description with summary of changes.

Reply:
@arpitbbhayani agreed, let's create an issue for now to keep track of it. This isn't super necessary at the moment to be fixed right away.

Sure, I'll go through the PR :)

Reply:
Yes a complete equality operator is more complex like the way the google did it. I've seen Joe Tsai gopher con where he stated that "using reflect.DeepEqual in tests is often a bad idea, that's why we open sourced go-cmp", May be we can take some ideas from them and include it in a separate repository **DiceDB Test Suite**.

Reply:
Have created an issue to track the changes.